docs/docs/how_to.md

```diff
@@ -0,0 +1,227 @@
+# How to program your robot
+
+## Controllers
+
+Each object in the simulation world is able to be controlled by a program.
+This program is called **Controller**. Each of the robots should have exactly
+one controller **Python3** program. There is an invisible referee object in the simulation,
+which takes care of controlling the game and checking the rules.
+
+The controllers are located in `controllers` directory. The name of the controller
+must be located in a subfolder with the same name and this name ought to be specified
+in `soccer.wbt` file.
+
+## Hello world, robot!
+
+We have prepared an example robot controllers, which can be found in `controllers`
+directory prefixed with `rcj_soccer_player`. Each robot is programmed the same way,
+meaning it rotates to face the ball and goes towards it afterwards.
+
+We decided to make the code reusable and put some common methods into
+`RCJSoccerRobot` class inside `rcj_soccer_robot.py` file.
+
+Let's put together a simple program on how to program the robot.
+
+```python
```

Comment:
@Adman this is quite a big jump, I though we'd go line by line or something like that. Is that missing by any chance?

Reply:
Yeah, line by line is explained below.